reo/scenario.py

```diff
@@ -317,29 +337,108 @@ def setup_pv(pv_dict, latitude, longitude, time_steps_per_hour):
+                ghpghx_post["longitude"] = inputs_dict["Site"]["longitude"]
+                # Only SpaceHeating portion of Heating Load gets served by GHP, unless allowed by can_serve_dhw
+                if ghpghx_post.get("heating_thermal_load_mmbtu_per_hr") in [None, []]:
+                    if not inputs_dict["Site"]["GHP"].get("can_serve_dhw"):
+                        ghpghx_post["heating_fuel_load_mmbtu_per_hr"] = lpbf_space.load_list
+                    else:
+                        ghpghx_post["heating_fuel_load_mmbtu_per_hr"] = [lpbf_space.load_list[i] + lpbf_dhw.load_list[i] for i in range(len(lpbf_space.load_list))]
+                if dfm.boiler is not None:
+                    ghpghx_post["existing_boiler_efficiency"] = dfm.boiler.boiler_efficiency #boiler.boiler_efficiency
+                else:
+                    ghpghx_post["existing_boiler_efficiency"] = 0.8
+                if ghpghx_post.get("cooling_thermal_load_ton") in [None, []]:
+                    ghpghx_post["cooling_thermal_load_ton"] = [kwt / TONHOUR_TO_KWHT for kwt in dfm.cooling_load.load_list] #lpct.load_list
+                client = TestApiClient()
+                # Update ground thermal conductivity based on climate zone if not user-input
+                if not ghpghx_post.get("ground_thermal_conductivity_btu_per_hr_ft_f"):
+                    k_by_zone = copy.deepcopy(ghp.GHPGHXInputs.ground_k_by_climate_zone)
+                    climate_zone = get_climate_zone(ghpghx_post["latitude"], ghpghx_post["longitude"])
+                    ghpghx_post["ground_thermal_conductivity_btu_per_hr_ft_f"] = k_by_zone[climate_zone]
+                # Call /ghpghx endpoint to size GHP and GHX
+                ghpghx_post_resp = client.post('/v1/ghpghx/', data=ghpghx_post)
+                ghpghx_post_resp_dict = json.loads(ghpghx_post_resp.content)
+                ghp_uuid = ghpghx_post_resp_dict.get('ghp_uuid')
+                ghpghx_results_url = "/v1/ghpghx/"+ghp_uuid+"/results/"
+                ghpghx_results_resp = client.get(ghpghx_results_url)  # same as doing ghpMakeResponse(ghp_uuid)
+                ghpghx_results_resp_dict = json.loads(ghpghx_results_resp.content)
+                ghpghx_response_list.append(ghpghx_results_resp_dict)
+                #json.dump(ghpghx_response_list, open("ghpghx_response.json", "w"))
+                ghp_option_list.append(ghp.GHPGHX(dfm=dfm,
+                                                    response=ghpghx_response_list[i],
+                                                    **inputs_dict["Site"]["GHP"]))
+            # Update GHPModel with created ghpghx_response
+            tmp = dict()
+            tmp['ghpghx_response'] = ghpghx_response_list
+            ModelManager.updateModel('GHPModel', tmp, run_uuid)
+            # Sleep to avoid calling julia_api for /job (reopt) too quickly after /ghpghx (only required locally and/or for debugging?)
+            time.sleep(10)
```

Comment:
Why is this long sleep needed?

Reply:
I think it's unnecessarily long, but while testing I was getting some julia_api connection errors when I didn't have any sleep. I thought it had to do with POSTing to the julia_api /job endpoint too soon after POSTing to the julia_api /ghpghx endpoint. I just commented out the sleep and I didn't get any connection errors, so maybe we don't need it. Just to be safe and if you're OK with it, I'm going to reduce it to 1 second.

Reply:
Actually I just realized I left a note in there that it was giving me a connection error when sequentially running multiple julia_api /ghpghx endpoints back-to-back in the for-loop for the number of ghpghx_inputs that were passed. Either way, I'm not getting that connection error now, so not sure. Maybe my computer was bogged down then.